mmv1/third_party/terraform/resources/resource_storage_bucket.go.erb

```diff
@@ -392,6 +393,10 @@ func resourceStorageBucketCreate(d *schema.ResourceData, meta interface{}) error
 
 	// Get the bucket and location
 	bucket := d.Get("name").(string)
+	bucketConstraint, _ := regexp.MatchString("[goog][a-z0-9][a-z0-9_.-]{1,220}[a-z0-9]", bucket)
```

Comment:
Given many of our previously succeeding test are now failing this validation, I am assuming this regex doesn't match the API's restrictions. 

For example a potential test name `tf-bucket-object-content-random1234` fails here

Reply:
Fixed it by adding a separate validator function. Since validation using regexp is slow used strings method as well